From 69a67e25469636173ed181c46eb3a6d96b955194 Mon Sep 17 00:00:00 2001
From: Nikita Enin <n.enin@samsung.com>
Date: Tue, 18 Oct 2022 19:30:09 +0300
Subject: [PATCH 077/128] [aten,axdimm] Reshaped axdimm embedding bag return
 tensor

Reshaped tensor to contain more information about batch
Tensor still flat and contiguous underneath

Resloves: AXDIMM-452

Signed-off-by: Nikita Enin <n.enin@samsung.com>
---
 aten/src/ATen/native/axdimm/ops/EmbeddingBag.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/aten/src/ATen/native/axdimm/ops/EmbeddingBag.cpp b/aten/src/ATen/native/axdimm/ops/EmbeddingBag.cpp
index d9255188..95e2bca3 100644
--- a/aten/src/ATen/native/axdimm/ops/EmbeddingBag.cpp
+++ b/aten/src/ATen/native/axdimm/ops/EmbeddingBag.cpp
@@ -82,7 +82,8 @@ at::Tensor embedding_bag_forward_only_axdimm(
 
   return torch::from_blob(
       psum_buf,
-      {minibatch_size * ten.get_num_tables(), ten.get_sparse_feature_size()},
+      {ten.get_num_tables(), minibatch_size, ten.get_sparse_feature_size()},
+      {minibatch_size * ten.get_sparse_feature_size(), ten.get_sparse_feature_size(), 1},
       [](void* data) { delete[] static_cast<uint8_t*>(data); },
       torch::TensorOptions().dtype(weights.dtype()).device(torch::kCPU));
 }
-- 
2.34.1

